packages/compass-connections/src/stores/connections-store.ts

```diff
@@ -283,6 +321,20 @@ export function useConnections({
         ...cloneDeep(connectionInfoToBeSaved),
         lastUsed: new Date(),
       });
+
+      // remove the oldest recent connection if are adding a new one and
+      // there are already MAX_RECENT_CONNECTIONS_LENGTH recents.
+      // NOTE: there are edge cases that may lead to more than MAX_RECENT_CONNECTIONS_LENGTH
+      // to be saved (ie. concurrent run of Compass),
+      // however we accept it as long as the list of
+      // recents won't grow indefinitely.
+      if (
+        !connectionInfoToBeSaved.favorite &&
+        !connectionInfoToBeSaved.lastUsed &&
+        recentConnections.length >= MAX_RECENT_CONNECTIONS_LENGTH
+      ) {
+        await removeConnection(recentConnections[recentConnections.length - 1]);
```

Comment:
NOTE: this technically abuses a little the fact that `recentConnections` are already ordered in the right way. This may change, for example, if we introduce some way to reorder or filter that list dynamically .. not really something we are planning for now, but I could easily extract a `getRecentsByLastUsed(connections)` and use it both here and in `useMemo` instead.

Let me know if you have some thoughts about it.